fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -928,6 +928,14 @@ private PartitionColumnFilter createPartitionFilter(SlotDescriptor desc, List<Ex
                 if (!inPredicate.isLiteralChildren() || inPredicate.isNotIn()) {
                     continue;
                 }
+                if ((inPredicate.getChild(0) instanceof LiteralExpr)
```

Comment:
you can use `(inPredicate.getChild(0).unwrapExpr(false) instanceof LiteralExpr)`